caffe2/contrib/aten/gen_op.py

```diff
@@ -278,7 +278,7 @@ def find_factory_methods(decls):
                     # first tensor input is used to define the output type.
                     defined_inferred_type = True
                     env['statements'].append(
-                        'auto inferred_type = &({}.type());'.format(
+                        'auto inferred_type = &(static_cast<at::TypeExtendedInterface&>({}.type()));'.format(
```

Comment:
see above, can this just call at::getType or similar (if we always return TypeExtendedInterface) from Context.